src/main/java/org/wso2/siddhi/plugins/idea/psi/scopes/ParameterContainer.java

```diff
@@ -0,0 +1,26 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.siddhi.plugins.idea.psi.scopes;
+
+import org.antlr.jetbrains.adaptor.psi.ScopeNode;
+
+/**
+ * Indicates that the scope can contain parameters.
+ */
+public interface ParameterContainer extends ScopeNode {
```

Comment:
Do you need scopes?

Reply:
Planned to use in the future in case of using with parameters in siddhi definitions.